datatypes/src/primitives/time_interval.rs

```diff
@@ -188,7 +188,9 @@ impl TimeInterval {
     /// ```
     ///
     pub fn intersects(&self, other: &Self) -> bool {
-        self.start < other.end && self.end > other.start
+        other == self
+            || value_in_range(self.start, other.start, other.end)
+            || value_in_range(other.start, self.start, self.end)
```

Comment:
I guess changing this method should come with a test. Otherwise, I could change it back and nothing complains.

Reply:
yes we need more tests for this

Reply:
e414046